pkg/reconcile/pipeline/api.go

```diff
@@ -22,6 +24,21 @@ type Pipeline interface {
 	Process(binding interface{}) (bool, error)
 }
 
+// A container-like object.  EnvFrom is optional; all other fields are required.
+type MetaContainer struct {
+	Name        string
+	Env         []string
+	EnvFrom     []string
+	VolumeMount []string
```

Comment:
- `Env` and `VolumeMount` are governed by restricted jsonpaths in the spec.  By design, we can convert those to `[]string`, so let's do that instead of passing around jsonpath objects we need to re-evaluate.
- `Name` is also a restricted jsonpath, but since we never modify the name of a container we're injecting into, we can read it once and pass that around instead.
- `EnvFrom` is always going to be `[]{"envFrom"}`, since this isn't accessible from `ClusterWorkloadResourceMapping` resources.  We need to be able to project into `envFrom` fields for compatability with the `coreos.com` API, since this is how we currently do environment variable projection.